src/lib/getDataFromExternalSources.ts

```diff
@@ -51,53 +54,92 @@ export const getDataFromExternalSources = (
       return dataEntry;
     }
 
-    // At this stage, value should be of type jsonurl, verifiableuri or asseturl
-    if (typeof dataEntry.value === 'string') {
-      console.error(
-        `Value of key: ${dataEntry.name} (${dataEntry.value}) is string but valueContent is: ${schemaElement.valueContent}. Expected type should be object with url key.`,
-      );
-      return dataEntry;
-    }
-
-    if (!dataEntry.value) {
-      return dataEntry;
-    }
+    try {
+      // At this stage, value should be of type jsonurl, verifiableuri or asseturl
+      if (typeof dataEntry.value === 'string') {
+        throw new Error(
+          `Value of key: ${dataEntry.name} (${dataEntry.value}) is string but valueContent is: ${schemaElement.valueContent}. Expected type should be object with url key.`,
+        );
+      }
 
-    if (Array.isArray(dataEntry.value)) {
-      console.error(
-        `Value of key: ${dataEntry.name} (${dataEntry.value}) is string[] but valueContent is: ${schemaElement.valueContent}. Expected type should be object with url key.`,
-      );
-      return dataEntry;
-    }
+      if (!dataEntry.value) {
+        throw new Error(`Value of key: ${dataEntry.name} is empty`);
+      }
 
-    const urlDataWithHash = dataEntry.value; // Type URLDataWithHash
+      if (Array.isArray(dataEntry.value)) {
+        throw new Error(
+          `Value of key: ${dataEntry.name} (${dataEntry.value}) is string[] but valueContent is: ${schemaElement.valueContent}. Expected type should be object with url key.`,
+        );
+      }
 
-    let receivedData;
-    try {
-      const { url } = patchIPFSUrlsIfApplicable(urlDataWithHash, ipfsGateway);
+      const urlDataWithHash: URLDataWithHash =
+        dataEntry.value as URLDataWithHash; // Type URLDataWithHash
 
-      receivedData = await fetch(url).then(async (response) => {
-        if (!response.ok) {
-          return undefined;
+      const { url } = patchIPFSUrlsIfApplicable(
```

Comment:
could this destructuring fail?

Reply:
Checking, but I don't think so.

Reply:
No unless the data is already null ```js
export function patchIPFSUrlsIfApplicable(receivedData, ipfsGateway) {
    if (receivedData &&
        receivedData.url &&
        receivedData.url.indexOf('ipfs://') !== -1) {
        return Object.assign(Object.assign({}, receivedData), { url: receivedData.url.replace('ipfs://', ipfsGateway) });
    }
    return receivedData;
}
```. Not this is kind of problematic until the the data-providers url resolver is implemented to support arweave and other URLs. Having an ipfs:// constant inside of the code will be difficult to support going forward.